### ISSUE TITLE
Getting the error value without using forceDecode (exceptions)

### ISSUE THREAD

**Issue by aszenz** [michaljanocko/schemawax#9]
Hi,

Thanks for open sourcing this little library, I like the minimal api, it provides just the right basics.

Right now decode returns `null | T` , to get the error we need to use `forceDecode` and then catch the exception.

I'm wondering if we could have another function `validate` which doesn't throw an exception but instead returns a union of error and data.

```ts
validate(data: unknown) => {type: 'Error', message: String} | {type: 'Ok', data: T}
```
The above api is easy to pattern match using a switch case.

**Comment by michaljanocko** [comment]
I like this! I have a busy week at work though. I will get to this eventually in the second half of the week.

However, if you feel like you're welcome to make a PR. It's not hard!

**Comment by michaljanocko** [comment]
Closed with @aszenz nailing #10.
